From 651736ea42939ae67169d3429489ae9f304eae3c Mon Sep 17 00:00:00 2001
From: Max Ihlenfeldt <max@igalia.com>
Date: Fri, 8 Dec 2023 11:47:43 +0000
Subject: [PATCH] Don't pass unknown LLVM options

Because we compile and link with an older LLVM version than upstream, we
need to drop the following LLVM options that aren't available yet:
* split-threshold-for-reg-with-hint

Trying to pass them anyways will lead to an "Unknown command line
argument" error.

Upstream-Status: Inappropriate [specific to older versions of LLVM]
Signed-off-by: Max Ihlenfeldt <max@igalia.com>
---
 build/config/compiler/BUILD.gn | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 8951f6006c..cc300e7122 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -591,24 +591,6 @@ config("compiler") {
       }
     }

-    # TODO(crbug.com/40283598): This causes binary size growth and potentially
-    # other problems.
-    # TODO(crbug.com/40284925): This isn't supported by Cronet's mainline llvm version.
-    if (default_toolchain != "//build/toolchain/cros:target" &&
-        !llvm_android_mainline) {
-      cflags += [
-        "-mllvm",
-        "-split-threshold-for-reg-with-hint=0",
-      ]
-      if (use_thin_lto && is_a_target_toolchain) {
-        if (is_win) {
-          ldflags += [ "-mllvm:-split-threshold-for-reg-with-hint=0" ]
-        } else {
-          ldflags += [ "-Wl,-mllvm,-split-threshold-for-reg-with-hint=0" ]
-        }
-      }
-    }
-
     # TODO(crbug.com/40192287): Investigate why/if this should be needed.
     if (is_win) {
       cflags += [ "/clang:-ffp-contract=off" ]
